### PR TITLE
Require white space after pytorchbot

### PR DESCRIPTION
**Pull request by kit1980**
To prevent things like https://github.com/pytorch/pytorch/pull/133247#issuecomment-2284725996

### CODE REVIEW

**Review by clee2000 — APPROVED** [review]

**Review by atalman — APPROVED** [review]

File: torchci/lib/bot/cliParser.ts [hunk 1]
```diff
@@ -187,7 +187,7 @@ parser.add_argument("-h", "--help", {
   action: "store_true",
 });
 
-const botCommandPattern = new RegExp(/^ *@pytorch(merge|)bot.*$/m);
```
Comment (atalman) [inline torchci/lib/bot/cliParser.ts:190]: Please post test use case should we use ``\s`` to match the space instead ?

Comment (kit1980) [reply torchci/lib/bot/cliParser.ts:190]: > should we use \s to match the space instead

That would match tabs and line breaks also, we want just literal whitespace.

**Review by malfet — APPROVED** [review]

File: torchci/lib/bot/cliParser.ts [hunk 2]
```diff
@@ -187,7 +187,7 @@ parser.add_argument("-h", "--help", {
...
+const botCommandPattern = new RegExp(/^ *@pytorch(merge|)bot .*$/m);
```
Comment (malfet) [inline torchci/lib/bot/cliParser.ts:190]: Multiple spaces between at-mention and command should be OK as well, shouldn't it? (I.e. `@pytorchbot   help`)
```suggestion
const botCommandPattern = new RegExp(/^ *@pytorch(merge|)bot +.*$/m);
```

Comment (kit1980) [reply torchci/lib/bot/cliParser.ts:190]: No need for that, it's already handled by `.*`

Comment (ZainRizvi) [inline torchci/lib/bot/cliParser.ts:190]: Consider using a generic whitespace delimiter, which will work with tabs and other separators as well https://stackoverflow.com/a/47763239/21539

```suggestion
const botCommandPattern = new RegExp(/^ *@pytorch(merge|)bot\s.*$/m);
```

**Review by ZainRizvi — APPROVED** [review]
Nice

File: torchci/lib/bot/cliParser.ts (hunk 2 above)
Comment (kit1980) [reply torchci/lib/bot/cliParser.ts:190]: Andrey already commented that, but we don't want line breaks and other, we want literal space.

File: torchci/lib/bot/cliParser.ts (hunk 1 above)
Comment (kit1980) [reply torchci/lib/bot/cliParser.ts:190]: Because it mimics command line, you don't use line breaks in command line to separate argument.

**Pull request merged**